app/src/main/java/com/example/simbirsoft_android_practice/filter/FilterFragment.kt

```diff
@@ -75,30 +61,25 @@ class FilterFragment : Fragment(R.layout.fragment_filter) {
         }
     }
 
-    private fun loadCategoryData() {
-        showLoading()
-
+    private fun observeCategories() {
         viewLifecycleOwner.lifecycleScope.launch {
-            categoryRepository.getCategories()
-                .flowOn(Dispatchers.IO)
-                .map { list ->
-                    list.map { category: Category ->
-                        CategoryMapper.toFilterCategory(
-                            category,
-                            filterPrefs,
-                        )
-                    }
-                }
-                .catch { throwable ->
-                    Log.w(
-                        TAG_FILTER_FRAGMENT,
-                        "Flow exception: ${throwable.localizedMessage}",
-                        throwable,
-                    )
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                filterViewModel.categories.collect { categories ->
+                    filterAdapter.submitList(categories)
                 }
-                .collect { categories ->
-                    showData(categories)
+            }
+        }
+    }
+
+    private fun observeLoading() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                filterViewModel.loading.collect { isLoading ->
+                    binding.progressBarFilter.isVisible = isLoading
+                    binding.imageViewFilterApplySettings.isVisible = !isLoading
+                    binding.recyclerViewFilterItem.isVisible = !isLoading
```

Comment:
Если есть возможность, избегай логических отрицаний. Они делают логику сложнее
```suggestion
                    binding.imageViewFilterApplySettings.isGone = isLoading
                    binding.recyclerViewFilterItem.isGone = isLoading
```